assets/queries/cloudFormation/cmk_unencrypted_storage/query.rego

```diff
@@ -49,48 +49,43 @@ CxPolicy[result] {
 	}
 }
 
-CxPolicy[result] { #Resource with name containing DB and KmsKeyId is undefined
+CxPolicy[result] {
 	document := input.document[i]
 	resource := document.Resources[key]
-	not isDBType(resource.Type)
-	isDBName(key)
+	resource.Type == "AWS::Redshift::Cluster"
 	properties := resource.Properties
-	object.get(properties, "KmsKeyId", "undefined") == "undefined"
+	object.get(properties, "Encrypted", "undefined") == "undefined"
+
 	result := {
 		"documentId": input.document[i].id,
 		"searchKey": sprintf("Resources.%s.Properties", [key]),
 		"issueType": "MissingAttribute",
-		"keyExpectedValue": sprintf("Resources.%s.Properties.KmsKeyId should be defined with AWS-Managed CMK", [key]),
-		"keyActualValue": sprintf("Resources.%s.Properties.KmsKeyId is undefined", [key]),
+		"keyExpectedValue": sprintf("Resources.%s.Properties.Encrypted should be defined", [key]),
+		"keyActualValue": sprintf("Resources.%s.Properties.Encrypted is undefined", [key]),
 	}
 }
 
 CxPolicy[result] {
 	document := input.document[i]
 	resource := document.Resources[key]
-	isDBName(key)
-	properties := resource.Properties
-	object.get(properties, "StorageEncrypted", "undefined") == "undefined"
+	resource.Type == "AWS::Redshift::Cluster"
+	properties := resource.Properties.Encrypted == false
 
 	result := {
 		"documentId": input.document[i].id,
-		"searchKey": sprintf("Resources.%s.Properties", [key]),
-		"issueType": "MissingAttribute",
-		"keyExpectedValue": sprintf("Resources.%s.Properties.StorageEncrypted should be defined", [key]),
-		"keyActualValue": sprintf("Resources.%s.Properties.StorageEncrypted is undefined", [key]),
+		"searchKey": sprintf("Resources.%s.Properties.Encrypted", [key]),
+		"issueType": "IncorrectValue",
+		"keyExpectedValue": sprintf("Resources.%s.Properties.Encrypted should be true", [key]),
+		"keyActualValue": sprintf("Resources.%s.Properties.Encrypted is false", [key]),
 	}
 }
 
-isDBName(name) = result {
-	possibleDBs := {"couchbase", "riak", "redis", "hbase", "Oracle", "SAP Hana", "Postgres", "cassandra", "hadoop", "Mongo", "Neo4j", "DB", "SQL"}
-	result := contains(name, possibleDBs[_])
-} else {
-	result := false
+checkTypeS(type){
```

Comment:
why have this checkTypeS and checkTypeK?

Reply:
To check for the different attributes, `checkTypeS` is for the types in which the existence of "StorageEncrypted" is to be verified and `checkTypeK` is for when "KmsKeyId" is to be verified.

Reply:
okay, then rename the function to have a more explicit name, checkTypeStorage

Reply:
Done